hawkular-inventory-api/src/main/java/org/hawkular/inventory/base/Util.java

```diff
@@ -352,14 +356,15 @@ public static Query extendTo(TraversalContext<?, ?> context, Path path) {
                 }
                 //not only the entity, but also its relationships are going to disappear
                 deletedRels.addAll(context.backend.getRelationships(e, both));
-            });
 
-            if (context.backend.hasRelationship(entity, outgoing, defines.name())) {
-                verticesToDeleteThatDefineSomething.add(entity);
-            } else {
-                deleted.add(entity);
-            }
-            deletedRels.addAll(context.backend.getRelationships(entity, both));
+                context.backend.getRelationships(e, outgoing, hasData.name()).forEach(rel -> {
+                    dataToBeDeleted.add(context.backend.getRelationshipTarget(rel));
+                });
+            };
+
+            categorizer.accept(entity);
```

Comment:
Yes, so that the `entity` ends up in the `deleted` or `verticesToDeleteThatDefineSomething` sets.